src/main/java/dk/cngroup/lentils/service/CustomUserDetailsService.java

```diff
@@ -0,0 +1,27 @@
+package dk.cngroup.lentils.service;
+
+import dk.cngroup.lentils.security.CustomUserDetails;
+import dk.cngroup.lentils.entity.User;
+import dk.cngroup.lentils.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(final String username) throws UsernameNotFoundException {
+        Optional<User> optionalUser = userRepository.findByUsername(username);
```

Comment:
Pouze komentář: Tady jsem se nad tím tak zamyslel na chvíli, možná by to šlo napsat ještě nějak takhle:
```
User user = userRepository.findByUsername(username)
            .orElseThrow(() -> new UsernameNotFoundException("Uživatel neexistuje."));

return new CustomUserDetails(user);
```
Pak je tam to orElseThrow tak jakoby více v "přirozené řeči". Nevím co je čitelnější.

Ale jinak OK, kdy bude merge? :D